keps/1618-optional-gc-of-workloads/README.md

```diff
@@ -0,0 +1,441 @@
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+[//]: # (##### Prerequisite testing updates)
+
+[//]: # ()
+[//]: # (<!--)
+
+[//]: # (Based on reviewers feedback describe what additional tests need to be added prior)
+
+[//]: # (implementing this enhancement to ensure the enhancements have also solid foundations.)
+
+[//]: # (-->)
+
+#### Unit Tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+The following new unit tests or modifications to existing ones were added to test 
+the new functionality:
+
+- `apis/config/v1beta1/defaults_test.go`
+  - updated existing tests with `defaultObjectRetentionPolicies` which sets 
+  `FinishedWorkloadRetention` to nil which ensures that the default 
+  scenario maintains backward compatibility and no retention settings are set.
+  - new unit test: `set object retention policy for finished workloads` which
+  checks if provided can configuration overrides default values.
+- `pkg/config/config_test.go`
+  - updated existing tests with `defaultObjectRetentionPolicies` set to nil 
+  which tests if for existing tests the default behavior maintains backward
+  compatibility and results in no retention settings.
+  - new unit test: `objectRetentionPolicies config` which sets some example 
+  retention policy and then checks if it can be retrieved.
+- `pkg/controller/core/workload_controller_test.go`
+  - added following new tests:
+    - `shouldn't delete the workload because, object retention not configured`
+    - `shouldn't try to delete the workload (no event emitted) because it is already being deleted by kubernetes, object retention configured`
+    - `shouldn't try to delete the workload because the retention period hasn't elapsed yet, object retention configured`
+    - `should delete the workload because the retention period has elapsed, object retention configured`
+    - `should delete the workload immediately after it completed because the retention period is set to 0s`
+
+#### Integration tests
+
+<!--
```

Comment:
I think a single scenario to verify that workload is deleted for a finished Job when the retention is configured would be very helpful. (maybe there are more scenarios, but I believe they could be figured out during implementation).

Reply:
Added the test to the original PR and mentioned it here.